src/Mvc/Mvc.Core/src/ModelBinding/FormFileValueProviderFactory.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    /// <summary>
+    /// A <see cref="IValueProviderFactory"/> for <see cref="FormValueProvider"/>.
+    /// </summary>
+    public sealed class FormFileValueProviderFactory : IValueProviderFactory
+    {
+        /// <inheritdoc />
+        public Task CreateValueProviderAsync(ValueProviderFactoryContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var request = context.ActionContext.HttpContext.Request;
+            if (request.HasFormContentType)
+            {
+                // Allocating a Task only when the body is multipart form.
```

Comment:
This comment isn't super duper accurate. `HasFormContentType` will be true for both form content types, not just multipart.

Reply:
Ahh, I changed the code, forgot to update the comment.